dmp_file.c

```diff
@@ -5,10 +5,17 @@
 #include "dmp_file.h"
 #include "tools.h"
 
-int dmp_file_load(struct dmp_file *f, uint8_t *data, size_t data_len, int system) {
+#define SYSTEM_YM2612_OPN		0x02
```

Comment:
I am not sure if these constants belong in this file or even in this position (I like to place them at the top of the file). But since they are used in a field that's defined in the struct, and the struct is defined in the header file, it's better if we move these inside the header file.

Reply:
The DMP format changed with each iteration, making it hard to keep track to write proper importer/exporter libraries as you have to consider each and every version if you want to make it properly. The mentioned constants are for version 11 or above that now also support OPM patches in addition to OPN, so a new header field indicating the target chip/system was added with the two possible values 2 and 8 relevant for us. Before that, the system was always OPN.
I agree with the placement of the defines though, I moved them to the header now.